src/metabase/pulse.clj

```diff
@@ -537,85 +535,16 @@
-  :type :integer
-  :default 7
-  :on-change reconfigure-retrying)
-
-(defsetting notification-retry-initial-interval
-  (deferred-tru "The initial retry delay in milliseconds when delivering notifications.")
-  :type :integer
-  :default 500
-  :on-change reconfigure-retrying)
-
-(defsetting notification-retry-multiplier
-  (deferred-tru "The delay multiplier between attempts to deliver a single notification.")
-  :type :double
-  :default 2.0
-  :on-change reconfigure-retrying)
-
-(defsetting notification-retry-randomization-factor
-  (deferred-tru "The randomization factor of the retry delay when delivering notifications.")
-  :type :double
-  :default 0.1
-  :on-change reconfigure-retrying)
-
-(defsetting notification-retry-max-interval-millis
-  (deferred-tru "The maximum delay between attempts to deliver a single notification.")
-  :type :integer
-  :default 30000
-  :on-change reconfigure-retrying)
-
-(defn- retry-configuration []
-  (cond-> {:max-attempts (notification-retry-max-attempts)
-           :initial-interval-millis (notification-retry-initial-interval)
-           :multiplier (notification-retry-multiplier)
-           :randomization-factor (notification-retry-randomization-factor)
-           :max-interval-millis (notification-retry-max-interval-millis)}
-    (or config/is-dev? config/is-test?) (assoc :max-attempts 1)))
-
-(defn- make-retry-state
-  "Returns a notification sender wrapping [[send-notifications!]] retrying
-  according to `retry-configuration`."
-  []
-  (let [retry (retry/random-exponential-backoff-retry "send-notification-retry"
-                                                      (retry-configuration))]
-    {:retry retry
-     :sender (retry/decorate send-notification! retry)}))
-
-(defonce
-  ^{:private true
-    :doc "Stores the current retry state. Updated whenever the notification
-  retry settings change.
-  It starts with value `nil` but is set whenever the settings change or when
-  the first call with retry is made. (See #22790 for more details.)"}
-  retry-state
-  (atom nil))
-
-(defn- reconfigure-retrying [_old-value _new-value]
-  (log/info (trs "Reconfiguring notification sender"))
-  (reset! retry-state (make-retry-state)))
+    (email/send-message-or-throw! {:subject      subject
+                                   :recipients   recipients
+                                   :message-type message-type
+                                   :message      message
+                                   :bcc?         (email/bcc-enabled?)})))
 
 (defn- send-notification-retrying!
-  "Like [[send-notification!]] but retries sending on errors according
-  to the retry settings."
+  "Like [[send-notification!]] but retries sending on errors according to the retry settings."
   [& args]
-  (when-not @retry-state
-    (compare-and-set! retry-state nil (make-retry-state)))
-  (apply (:sender @retry-state) args))
+  (apply (retry/decorate send-notification!) args))
```

Comment:
It looks like this is creating a new decorated version of `send-notification!` each time `send-notification-retrying!` is called, which also creates a new `Retry` instance each time. I don't think that should be necessary — we should only create the `Retry` instance once for a given set of settings. That's what the previous `retry-state` atom was for.

Was there a reason for getting rid of that mechanism?